### PR TITLE
[18186] Update endpoint qos XML schema

### PR DESCRIPTION
**Pull request by JesusPoderoso**
## Description

Previous configuration would allow to include data writer QoS in data reader QoS XML configuration files. Those QoS from writer do not exist in the reader configuration.

The implemented solution is based on duplicating the QoS schema, keeping the complete one for the data writer entity, and the  new one (without `disable_heartbeat_piggyback` and `publishMode` QoS) has been assigned to the data reader.

XML test configuration files has been updated and test has passed locally.

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [x] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 

**N/A** Any new/modified methods have been properly documented using Doxygen. 

**N/A** Changes are ABI compatible. 

**N/A** Changes are API compatible. 

**N/A** New feature has been added to the `versions.md` file (if applicable).

- [x] New feature has been documented/Current behavior is correctly described in the documentation. 
    

**N/A** Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: resources/xsd/fastRTPS_profiles.xsd [hunk 1]
```diff
@@ -172,7 +172,7 @@
     <xs:complexType name="publisherProfileType">
         <xs:all>
             <xs:element name="topic" type="topicElementType" minOccurs="0" maxOccurs="1"/>
-            <xs:element name="qos" type="QosPoliciesType" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="qos" type="publisherQosPoliciesType" minOccurs="0" maxOccurs="1"/>
```
Comment (JLBuenoLopez-eProsima) [inline resources/xsd/fastRTPS_profiles.xsd:175]: I suggest using DDS terminology: `dataWriterQosPoliciesType` and `dataReaderQoSPoliciesType`. Does it make sense to have the common Qos Policies in the previous type and define only a new QoS type with the specific ones that apply only to DataWriters? Can this be done?

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]

**Comment by JLBuenoLopez-eProsima** [comment]
CI failure is unrelated as it is also occurring in MacOS nightly runs.

File: resources/xsd/fastRTPS_profiles.xsd (hunk 1 above)
Comment (JesusPoderoso) [reply resources/xsd/fastRTPS_profiles.xsd:175]: Yes, it makes sense

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM

**Comment by JesusPoderoso** [comment]
@richiprosima please test this

**Pull request merged**